kernel/src/process.rs

```diff
@@ -355,7 +355,9 @@ pub trait ProcessType {
     ///
     /// This will return `None` if the process is inactive and the grant region
     /// cannot be used.
-    unsafe fn grant_ptr(&self, grant_num: usize) -> Option<*mut *mut u8>;
+    ///
+    /// Caution: Pointer may be null if the grant has not yet been allocated.
+    fn grant_ptr(&self, grant_num: usize) -> Option<&mut *mut u8>;
```

Comment:
I may have missed something, but I don't see any of the callers changing the pointer. Shouldn't this function return an `Option<*mut u8>`?

Reply:
Nevermind, it's written by `*grant_ptr = new_ctr_ptr as *mut u8;`.

Reply:
I actually did that at first because I missed that case on my first skim, then had to back out the changes 🤦‍♂